src/ts_catalog/chunk_data_node.c

```diff
@@ -353,3 +413,221 @@ ts_chunk_data_nodes_scan_iterator_set_node_name(ScanIterator *it, const char *no
 								   F_NAMEEQ,
 								   CStringGetDatum(node_name));
 }
+
+bool
+ts_chunk_match_data_node_by_server(const Chunk *chunk, const ForeignServer *server)
```

Comment:
Hm.. curious if we should move this function to `src/chunk.h` we have function `ts_chunk_has_data_node()` which is very alike  or you can match by name instead

Reply:
Ok, done.